### PR TITLE
Describe order of top-level structures

### LINKED ISSUE #80: Describe order of all top-level structures inside a file
Currently we have rule 3.2 regarding file header and recommendation 3.2 regarding order of declarations inside class or object. Need to add requirements on declarations inside a file as a whole.

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Added rule 3.1.5 description

This pull request closes #80

## Actions checklist
* [ ] Implemented Rule, added Warnings
* [ ] Added tests on checks
* [ ] Added tests on fixers
* [ ] Updated diktat-analysis.yml
* [ ] Updated available-rules.md

### CODE REVIEW

File: info/guide/guide-chapter-3.md [hunk 1]
```diff
@@ -77,6 +77,60 @@ If the classes are meant to be used externally, and are not referenced inside th
 **Exception:**
 All variants of a `(private) val` logger should be placed at the beginning of the class (`(private) val log`, `LOG`, `logger`, etc.).
 
+#### <a name="r3.1.5"></a> 3.1.5 Order of declaration of top-level code structures
+Kotlin allows several top-level declaration types: classes, objects, interfaces, properties and functions.
+When declaring more than one class or zero classes (e.g. only functions), as per rule 2.2.1, you should document the whole file in the header KDoc.
```
Comment (akuleshov7) [inline info/guide/guide-chapter-3.md:82]: link to rule

File: info/guide/guide-chapter-3.md [hunk 2]
```diff
@@ -77,6 +77,60 @@ If the classes are meant to be used externally, and are not referenced inside th
...
+When declaring top-level structures, keep the following order:
+1. Top-level public constants (`const val`)
+2. Top-level public properties
+   1.1 `val`
+   1.2 `lateinit var`
+   1.3 `var`
+3. Public classes
+4. Public extension functions
+5. Public functions
+6. Private constants and properties, following same order as in 1 and 2
```
Comment (akuleshov7) [inline info/guide/guide-chapter-3.md:92]: internal also here?

File: info/guide/guide-chapter-3.md [hunk 3]
```diff
@@ -77,6 +77,60 @@ If the classes are meant to be used externally, and are not referenced inside th
...
+7. Private classes
+8. Private extension functions
+9. Private functions
+
+**Note**:
+Extension functions, that use the class or interface declared in the same file (accept parameters of this type or return this type),
```
Comment (akuleshov7) [inline info/guide/guide-chapter-3.md:98]: please create a feature to check that no one creates extension function in the same file with the class

File: info/guide/guide-chapter-3.md [hunk 4]
```diff
@@ -77,6 +77,60 @@ If the classes are meant to be used externally, and are not referenced inside th
...
+
+fun Other.asExample(): Example { /*...*/ }
+
+private class Internal
+
+private fun Other.asInternal(): Internal
+```
+
+**Note**:
+kotlin scripts (.kts) allow arbitrary code to be placed on the top level. When writing kotlin scripts, you should first declare all properties, classes
```
Comment (akuleshov7) [inline info/guide/guide-chapter-3.md:121]: please create the following issue: 
1) If kts script contains other functions except run code, then need to wrap it in run {}

**Review by akuleshov7 — CHANGES_REQUESTED** [review]
please create rules for it (better milestone)

File: info/guide/guide-chapter-3.md (hunk 3 above)
Comment (petertrr) [reply info/guide/guide-chapter-3.md:98]: #731

File: info/guide/guide-chapter-3.md (hunk 4 above)
Comment (petertrr) [reply info/guide/guide-chapter-3.md:121]: #732

**Comment by petertrr** [comment]
@akuleshov7 as discussed, removed unnecessary complications from rule, now we have just 4 groups of declarations

**Review by akuleshov7 — APPROVED** [review]
lgtm

**Pull request merged**